formats/hocon/src/main/kotlin/kotlinx/serialization/hocon/Hocon.kt

```diff
@@ -161,8 +161,12 @@ public sealed class Hocon(
         private fun composeName(parentName: String, childName: String) =
             if (parentName.isEmpty()) childName else "$parentName.$childName"
 
-        override fun SerialDescriptor.getTag(index: Int): String =
-            composeName(currentTagOrNull.orEmpty(), getConventionElementName(index, useConfigNamingConvention))
+        override fun SerialDescriptor.getTag(index: Int): String {
+            return if (!isPolymorph) composeName(
+                currentTagOrNull.orEmpty(),
+                getConventionElementName(index, useConfigNamingConvention)
+            ) else getElementName(index)
```

Comment:
Shouldn't we also apply `getConventionElementName` here?

Reply:
Didn't think of that, yeah it definitely should!